### PR TITLE
docs: Segregated SSO Docs + Added Auto-Assign Permissions Feature

### PR DESCRIPTION
**Pull request by ashokdevtron**
- Now, there will be separate docs for SSO providers
- SSO docs will now have a short guidance to fetch client ID and secret
- Auto-assign feature covered in Microsoft and LDAP docs
- Amended and linked other docs (except old snapshots - will be done in another sprint)
- Added enterprise tagging and RBAC

### CODE REVIEW

File: docs/user-guide/global-configurations/authorization/user-access.md [hunk 1]
```diff
@@ -188,6 +188,10 @@ Click `Save`once you have configured all the required permissions for the users.
 
 ### Edit User Permissions
 
+{% hint style="warning" %}
+You cannot edit user permissions if 'auto-assign permission' is enabled in the SSO configuration of either [Microsoft](../authorization/sso/microsoft.md#auto-assign-permissions) or [LDAP](../authorization/sso/ldap.md#auto-assign-permissions).
```
Comment (uxarya-d) [inline docs/user-guide/global-configurations/authorization/user-access.md:192]: Rephrase suggestion: Direct user permissions cannot be edited if you're using LDAP/Microsoft for SSO and 'auto-assign permission' is enabled. Permissions can only be managed via permission groups in such a scenario. //Hyperlink to relevant section in permission groups?

**Review by uxarya-d — CHANGES_REQUESTED** [review]
Added review comments.

File: docs/user-guide/global-configurations/authorization/permission-groups.md [hunk 2]
```diff
@@ -16,7 +16,7 @@ Enter the `Group Name` and `Description`.
 
 ![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/permission-group/permission-group-2.png)
 
-Assign the permissions of groups to users to manage access for:
+You can either grant [super-admin](../../global-configurations/authorization/user-access.md#role-based-access-levels) permission to a user group or specific permissions to manage access for:
 
    * [Devtron Apps](#devtron-apps-permissions)
    * [Helm Apps](#helm-apps-permissions)
```
[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/user-guide/global-configurations/authorization/permission-groups.md:24]: Added **Jobs** in this doc

File: docs/user-guide/global-configurations/authorization/sso/github.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
+# GitHub
+
+## Sample Configuration
+
+![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/sso-login-service/github.jpg)
```
[the remark this answers is not in the document]
Comment (ashokdevtron) [reply docs/user-guide/global-configurations/authorization/sso/github.md:5]: Added redirect URI, but removed the infobar. The doc is optimized now (please check the attached snapshot).
image

**Comment by ashokdevtron** [comment]
> Added review comments.

Incorporated all the feedback. Thanks!

File: docs/user-guide/global-configurations/authorization/permission-groups.md [hunk 4]
```diff
@@ -62,25 +67,47 @@ Provide the information in the following fields:
...
+| **Job Name** | Select the specific job name or all jobs from the drop-down list.<br>Note: If you select `All Jobs` option, then the user gets access to all the current jobs including any new job which gets associated with the project later.</br>  |
+| **Workflow**  | Select the specific workflow or all workflows from the drop-down list.<br>Note: If you select `All Workflows` option, then the user gets access to all the current workflows including any new workflow which gets associated with the project later.</br> |
+| **Environment** | Select the specific environment or all environments from the drop-down list.<br>Note: If you select `All environments` option, then the user gets access to all the current environments including any new environment which gets associated with the project later.</br> |
+| **Role**  | Select one of the [roles](#role-based-access-levels) to which you want to give permission to the user:<ul><li>`View only`</li></ul> <ul><li>`Run job`</li></ul><ul><li>`Admin`</li></ul>  |
```
Comment (abhibhaw) [inline docs/user-guide/global-configurations/authorization/permission-groups.md:86]: Approver roles?

File: docs/user-guide/global-configurations/authorization/sso/google.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+* clientID
+
+* clientSecret
+
+    ![Fetching Client ID and Secret](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/sso-login-service/secret/google-id-secret.jpg)
+
+
+### Values to Provide
+
+* redirectURI (provided in SSO Login Services by Devtron)
```
Comment (abhibhaw) [inline docs/user-guide/global-configurations/authorization/sso/google.md:24]: Hosted domain and its use is missing, also the attached screenshot has a definite redirect url, hope its not confusing, how to get a redirect URL? that can also be added

Comment (abhibhaw) [reply docs/user-guide/global-configurations/authorization/sso/google.md:24]: Maybe highlight the redirect URL which we show above as a person can use that directly

File: docs/user-guide/global-configurations/authorization/sso/microsoft.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+## Reference
+
+[View Microsoft Documentation](https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-register-app)
+
+---
+
+## Auto-assign Permissions [![](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/elements/EnterpriseTag.svg)](https://devtron.ai/pricing)
+
+{% hint style="info" %}
+Only a super-admin can enable this feature. Make sure to add tenantID in the SSO configuration field without fail.
```
Comment (abhibhaw) [inline docs/user-guide/global-configurations/authorization/sso/microsoft.md:42]: Only superadmin can configure anything related to SSO, not just this. Also who's superadmin if no SSO is configured? :)

File: docs/user-guide/global-configurations/authorization/sso/microsoft.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+{% endhint %}
+
+Since Microsoft supports <a href="https://learn.microsoft.com/en-us/windows-server/identity/ad-ds/get-started/virtual-dc/active-directory-domain-services-overview" target="_blank">Active Directory (AD)</a>
+, this feature further simplifies the onboarding process of organizations having a large headcount of users. It also eliminates repetitive permission assignment by automatically mapping your Azure AD groups to Devtron's [Permission Groups](../permission-groups.md) during single sign-on (SSO) login.
+
+![Enabling Permission Auto-assignment](https://devtron-public-asset.s3.us-east-2.amazonaws.com/images/global-configurations/sso-login-service/secret/auto-grant.jpg)
+
+In other words, if you have created groups within your active directory representing different roles or access levels, you can use the same group name to create a permission group in Devtron. When users belonging to a specific group on your active directory log in to Devtron for the first time, they will automatically get the pre-decided permissions granted to them. This also means that you cannot change or add separate [permissions for a user](../user-access.md) mapped to a permission group.
```
Comment (abhibhaw) [inline docs/user-guide/global-configurations/authorization/sso/microsoft.md:50]: they will get permissions(of devtron, on devtron) attached with the permission groups. Any permission at AD won't be populated here, mentioning as the text has roles and access levels defined at AD mentioned

File: docs/user-guide/global-configurations/authorization/sso/microsoft.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+{% hint style="warning" %}
+SSO login will fail if there's any mismatch between the name of permission group created in Devtron with the group present in your active directory (or if the group is absent altogether).
```
Comment (abhibhaw) [inline docs/user-guide/global-configurations/authorization/sso/microsoft.md:53]: For auto assign group, its worth mentioning that all the users will be able to signin, if you want to stop any user on purpose, mark them inactive

**Review by abhibhaw — CHANGES_REQUESTED** [review]
Also we need to mention it somewhere that the next step after adding normal SSO at Devtron is to add user, else no one can login. Similarly for auto assign group, permission group must exist

File: docs/user-guide/global-configurations/authorization/permission-groups.md (hunk 4 above)
Comment (ashokdevtron) [reply docs/user-guide/global-configurations/authorization/permission-groups.md:86]: This is the **Jobs** section. There's no approver role (as far as I can see, please check the attached snapshot):
image

File: docs/user-guide/global-configurations/authorization/sso/google.md (hunk 5 above)
Comment (ashokdevtron) [reply docs/user-guide/global-configurations/authorization/sso/google.md:24]: Hosted domain part is covered in DexIDP documentation whose link we have already provided in our UI. It's practically not possible to cover every single configuration key-value pairs in our doc. Initial intention was to include the process of fetching Client ID and Secret correctly from the SSO provider (which is not covered by DexIDP).

Comment (ashokdevtron) [reply docs/user-guide/global-configurations/authorization/sso/google.md:24]: > Maybe highlight the redirect URL which we show above as a person can use that directly

Done...added the snapshots

File: docs/user-guide/global-configurations/authorization/sso/microsoft.md (hunk 8 above)
Comment (ashokdevtron) [reply docs/user-guide/global-configurations/authorization/sso/microsoft.md:55]: Thanks, noted. Will include this line when the feature is released.

**Comment by ashokdevtron** [comment]
> Also we need to mention it somewhere that the next step after adding normal SSO at Devtron is to add user, else no one can login. Similarly for auto assign group, permission group must exist

Sure. Added a short section in [sso-login](https://github.com/devtron-labs/devtron/blob/main/docs/user-guide/global-configurations/sso-login.md) doc.

**Review by abhibhaw — APPROVED** [review]
LGTM

**Review by uxarya-d — APPROVED** [review]

**Pull request merged**